src/pages/iou/request/step/IOURequestEditReportCommon.tsx

```diff
@@ -0,0 +1,100 @@
+import React, {useMemo} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {useOnyx} from 'react-native-onyx';
+import SelectionList from '@components/SelectionList';
+import type {ListItem} from '@components/SelectionList/types';
+import UserListItem from '@components/SelectionList/UserListItem';
+import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
+import useDebouncedState from '@hooks/useDebouncedState';
+import useLocalize from '@hooks/useLocalize';
+import Navigation from '@libs/Navigation/Navigation';
+import {getOutstandingReportsForUser} from '@libs/ReportUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Route} from '@src/ROUTES';
+import type {Report} from '@src/types/onyx';
+import mapOnyxCollectionItems from '@src/utils/mapOnyxCollectionItems';
+import StepScreenWrapper from './StepScreenWrapper';
+
+type ReportListItem = ListItem & {
+    /** reportID of the report */
+    value: string;
+};
+
+/**
+ * This function narrows down the data from Onyx to just the properties that we want to trigger a re-render of the component.
+ * This helps minimize re-rendering and makes the entire component more performant.
+ */
+const reportSelector = (report: OnyxEntry<Report>): OnyxEntry<Report> =>
+    report && {
+        ownerAccountID: report.ownerAccountID,
+        reportID: report.reportID,
+        policyID: report.policyID,
+        reportName: report.reportName,
+        stateNum: report.stateNum,
+        statusNum: report.statusNum,
+        type: report.type,
+    };
+
+type Props = {
+    backTo: Route | undefined;
+    transactionReport: OnyxEntry<Report>;
+    selectReport: (item: ReportListItem) => void;
+};
+
+function IOURequestEditReportCommon({backTo, transactionReport, selectReport}: Props) {
+    const {translate} = useLocalize();
+    const [allReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {selector: (c) => mapOnyxCollectionItems(c, reportSelector), canBeMissing: true});
```

Comment:
I know this is not your code, but what is `c`? Can we use a better variable name in this case?